v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/SourceDbToSpanner.java

```diff
@@ -102,13 +102,18 @@ static PipelineResult run(SourceDbToSpannerOptions options) {
     SpannerConfig spannerConfig = createSpannerConfig(options);
 
     // Decide type and source of migration
-    if (options.getSourceConfigURL().startsWith("gs://")) {
+    // TODO(vardhanvthigle): Move this within pipelineController.
+    if (options.getSourceDbDialect().equals(SourceDbToSpannerOptions.CASSANDRA_SOURCE_DIALECT)) {
+      return PipelineController.executeCassandraMigration(options, pipeline, spannerConfig);
+    } else if (options.getSourceConfigURL().startsWith("gs://")) {
```

Comment:
suggestion - have a switch case for sources - and have this logic within the JDBC/mysql/pg source to make it clearer.

Reply:
Incorporated. Thanks.